templates/ikfast61_moveit_plugin_template.cpp

```diff
@@ -958,6 +1052,203 @@ bool IKFastKinematicsPlugin::getPositionIK(const geometry_msgs::Pose &ik_pose,
+    return false;
+  }
+
+  if(ik_poses.size() > 1)
+  {
+    ROS_ERROR("ik_poses contains multiple entries, only one is allowed");
+    result.kinematic_error = kinematics::KinematicErrors::MULTIPLE_TIPS_NOT_SUPPORTED;
+    return false;
+  }
+
+  if(ik_seed_state.size() < num_joints_)
+  {
+    ROS_ERROR_STREAM("ik_seed_state only has "<<ik_seed_state.size()<<" entries, this ikfast solver requires "<<num_joints_);
+    return false;
+  }
+
+  KDL::Frame frame;
+  tf::poseMsgToKDL(ik_poses[0],frame);
+
+  // solving ik
+  std::vector< IkSolutionList<IkReal> > solution_set;
+  IkSolutionList<IkReal> ik_solutions;
+  std::vector<double> vfree;
+  int numsol = 0;
+  std::vector<double> sampled_joint_vals;
+  if(!redundant_joint_indices_.empty())
+  {
+    // initializing from seed
+    sampled_joint_vals.push_back(ik_seed_state[redundant_joint_indices_[0]]);
+
+    // checking joint limits when using no discretization
+    if(options.discretization_method == kinematics::DiscretizationMethods::NO_DISCRETIZATION &&
+        joint_has_limits_vector_[redundant_joint_indices_.front()])
+    {
+      double joint_min = joint_min_vector_[redundant_joint_indices_.front()];
+      double joint_max = joint_max_vector_[redundant_joint_indices_.front()];
+
+      double jv = sampled_joint_vals[0];
+      if(!( (jv > (joint_min - LIMIT_TOLERANCE)) && (jv < (joint_max + LIMIT_TOLERANCE)) ))
+      {
+        result.kinematic_error = kinematics::KinematicErrors::IK_SEED_OUTSIDE_LIMITS;
+        ROS_ERROR_STREAM("ik seed is out of bounds");
+        return false;
+      }
+
+    }
+
+
+    // computing all solutions sets for each sampled value of the redundant joint
+    if(!sampleRedundantJoint(options.discretization_method,sampled_joint_vals))
+    {
+      result.kinematic_error = kinematics::KinematicErrors::UNSUPORTED_DISCRETIZATION_REQUESTED;
+      return false;
+    }
+
+    for(unsigned int i = 0; i < sampled_joint_vals.size(); i++)
+    {
+      vfree.clear();
+      vfree.push_back(sampled_joint_vals[i]);
+      numsol += solve(frame,vfree,ik_solutions);
+      solution_set.push_back(ik_solutions);
+    }
+  }
+  else
+  {
+    // computing for single solution set
+    numsol = solve(frame,vfree,ik_solutions);
+    solution_set.push_back(ik_solutions);
+  }
+
+  ROS_DEBUG_STREAM_NAMED("ikfast","Found " << numsol << " solutions from IKFast");
+  bool solutions_found = false;
+  std::stringstream ss;
+  if( numsol > 0 )
+  {
+    for(unsigned int r = 0; r < solution_set.size() ; r++)
+    {
+
+      ik_solutions = solution_set[r];
+      numsol = ik_solutions.GetNumSolutions();
+      for(int s = 0; s < numsol; ++s)
+      {
+        std::vector<double> sol;
+        getSolution(ik_solutions,s,sol);
+        ss.str("");
+        ss<<"[";
+        for(unsigned int i = 0 ; i < sol.size() ; i++)
+        {
+          ss<<sol[i]<<" ";
+        }
+        ss<<"]";
+        ROS_DEBUG_NAMED("ikfast","Sol %d: %s", s, ss.str().c_str());
```

Comment:
this is that inefficient section that was removed in another PR, right? we don't need this stringstream...